### PR TITLE
reading DMA BD sizes for bandwidth calculations

### PR DESCRIPTION
**Pull request by pgschuey**

### CODE REVIEW

**Comment by pgschuey** [comment]
@hackwa

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 1]
```diff
@@ -687,7 +734,7 @@ namespace xdp {
     auto it = mThreadCtrlMap.find(handle);
     if (it == mThreadCtrlMap.end())
       return;
-
+
```
Comment (keryell) [inline src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp:737]: Please configure your editor to avoid adding random useless spaces.

**Review by keryell — CHANGES_REQUESTED** [review]

**Comment by pgschuey** [comment]
please re-test

**Comment by hackwa** [comment]
Looks great! @pgschuey instead of reading all the BDs, one enhancement could be that we only read BDs that are valid? BD control has a bit that can be used to check if a bd is valid. This might help with information overload.

**Comment by kbhardwa-xilinx** [comment]
please re-test

**Comment by xilinxgitops** [comment]
retest this please

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 2]
```diff
@@ -547,6 +547,47 @@ namespace xdp {
...
+                                 XAIEGBL_MEM_DMABD2CTRL_LEN_LSB,  XAIEGBL_MEM_DMABD3CTRL_LEN_LSB,
+                                 XAIEGBL_MEM_DMABD4CTRL_LEN_LSB,  XAIEGBL_MEM_DMABD5CTRL_LEN_LSB,
+                                 XAIEGBL_MEM_DMABD6CTRL_LEN_LSB,  XAIEGBL_MEM_DMABD7CTRL_LEN_LSB};
+    uint32_t masks[NUM_BDS]   = {XAIEGBL_MEM_DMABD0CTRL_LEN_MASK, XAIEGBL_MEM_DMABD1CTRL_LEN_MASK,
+                                 XAIEGBL_MEM_DMABD2CTRL_LEN_MASK, XAIEGBL_MEM_DMABD3CTRL_LEN_MASK,
+                                 XAIEGBL_MEM_DMABD4CTRL_LEN_MASK, XAIEGBL_MEM_DMABD5CTRL_LEN_MASK,
+                                 XAIEGBL_MEM_DMABD6CTRL_LEN_MASK, XAIEGBL_MEM_DMABD7CTRL_LEN_MASK};
+
+    auto tileOffset = _XAie_GetTileAddr(aieDevInst, row + 1, column);
+    for (int bd=0; bd < NUM_BDS; ++bd) {
```
[the remark this answers is not in the document]
Comment (pgschuey) [reply src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp]: fixed

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp (hunk 1 above)
Comment (pgschuey) [reply src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp]: fixed

**Comment by pgschuey** [comment]
> Looks great! @pgschuey instead of reading all the BDs, one enhancement could be that we only read BDs that are valid? BD control has a bit that can be used to check if a bd is valid. This might help with information overload.

Added suggested validity check from @hackwa

**Review by jvillarre — APPROVED** [review]

**Pull request merged**